File: scripts/encryptUrl.py
from Crypto.Cipher import AES
import hashlib
import base64
import sys

if len(sys.argv) < 5:
	print 'Usage:\n\tphp encryptUrl.php <base url> <encrypted part> <encryption key> <encryption iv>'
	sys.exit(1)

BASE_URL = sys.argv[1]
ENC_URL = sys.argv[2]
ENC_KEY = sys.argv[3].decode('hex')
ENC_IV = sys.argv[4].decode('hex')
HASH_SIZE = 8

def pkcs7encode(text, blockSize = 16):
	val = blockSize - (len(text) % blockSize)
	for _ in xrange(val):
		text += chr(val)
	return text

m = hashlib.md5()
m.update(ENC_URL)
signedUrl = m.digest()[:HASH_SIZE] + ENC_URL
	
aes = AES.new(ENC_KEY, AES.MODE_CBC, ENC_IV)
encryptedUrl = base64.urlsafe_b64encode(aes.encrypt(pkcs7encode(signedUrl))).rstrip('=')

print BASE_URL + encryptedUrl

File: scripts/encryptUrl.php
<?php

if ($argc < 5)
{
	die("Usage:\n\tphp encryptUrl.php <base url> <encrypted part> <encryption key> <encryption iv>\n");
}

$BASE_URL = $argv[1];
$ENC_URL = $argv[2];
$ENC_KEY = pack("H*", $argv[3]);
$ENC_IV = pack("H*", $argv[4]);

$HASH_SIZE = 8;

// add signature
$signedUrl = substr(md5($ENC_URL, true), 0, $HASH_SIZE) . $ENC_URL;

// add PKCS#7 padding
$pad = 16 - (strlen($signedUrl) % 16);
$signedUrl .= str_repeat(chr($pad), $pad);

$encrypted = null;

// AES encrypt
if (function_exists('mcrypt_encrypt')) 
{
    $encrypted = mcrypt_encrypt(MCRYPT_RIJNDAEL_128, $ENC_KEY, $signedUrl, MCRYPT_MODE_CBC, $ENC_IV);
}
else if (function_exists('openssl_encrypt'))
{
    $encrypted = openssl_encrypt($signedUrl, 'aes-256-cbc', $ENC_KEY, OPENSSL_RAW_DATA | OPENSSL_NO_PADDING, $ENC_IV);
}
else
{
    die("need either mcrypt or openssl extension\n");
}

// base64 encrypt
$base64Encoded = rtrim(strtr(base64_encode($encrypted), '+/', '-_'), '=');

echo $BASE_URL . $base64Encoded . "\n";
